stdlib/public/core/LifetimeManager.swift

```diff
@@ -283,3 +283,40 @@ extension String {
 public func _copy<T>(_ value: T) -> T {
   copy value
 }
+
+/// Unsafely discard any lifetime dependency on the `dependent` argument. Return
+/// a value identical to `dependent` with a lifetime dependency on the caller's
+/// borrow scope of the `borrows` argument.
+///
+/// TODO: Remove @_unsafeNonescapableResult. Instead, the unsafe dependence
+/// should be expressed by a builtin that is hidden within the function body.
```

Comment:
This does not belong in the documentation comment; it needs to be moved to a regular comment within the function body. (To avoid interfering with limited documentation tools, regular comments cannot be placed in between a function and its documentation.)

Reply:
To be absolutely clear:
```
    let baseAddress = buffer.baseAddress
    let span = RawSpan(_unchecked: baseAddress, byteCount: buffer.count)
    self = _overrideLifetime(span, borrowing: buffer)
```
`overrideLifetime` changes the lifetime of `span` at the point it is evaluated. It does not change the lifetime of `baseAddress`. And `span` cannot escape its original dependence before it is passed to `overrideLifetime`. So, the `Span` initializer itself needs to have a well-formed lifetime dependence (it is not marked `@_unsafeNonescapableResult`)